distribution/docs/src/main/resources/_contents/_apis/migration-api-contents.adoc

```diff
@@ -0,0 +1,104 @@
+
+=== Migration API
+
+[NOTE]
+====
+This code is experimental. While this interface is functional and tested, it may change or be removed in a future version of the library.
+====
+
+${branding} currently has an experimental API for making bundles migratable. Interfaces in `platform/migration/platform-migratable-api` are
+used by the system to identify bundles that provide implementations for import (_coming soon_) and export operations.
+
+The `Migratable` API provides a mechanism for bundles to handle exporting data required to clone a ${branding} system.
+The migration process is meant to be flexible, so an implementation of `org.codice.ddf.migration.Migratable` can handle exporting data for a single bundle or groups of bundles such as applications.
+For example, the `org.codice.ddf.platform.migratable.impl.PlatformMigratable` handles exporting core system files for the ${ddf-platform} Application.
+Exporting configurations stored in `org.osgi.service.cm.ConfigurationAdmin` does not need to be handled by implementations of `org.codice.ddf.migration.Migratable` as all `ConfigurationAdmin` configurations are exported by `org.codice.ddf.configuration.admin.ConfigurationAminMigration`.
+An export operation can be performed through the ${command-console} or through the ${admin-console}. When a export operation is processed, the migration API
+will do a look-up for all registered OSGi services that are implementing `Migratable` and call their `export()` method.
+
+The services that implement one of the migratable interfaces will be called one at a time, in any order, and do not need to be thread safe. A bundle
+or a feature can have as many services implementing the interfaces as needed.
+
+==== The Migratable API Interfaces
+
+. `org.codice.ddf.migration.Migratable`
+. `org.codice.ddf.migration.AbstractMigratable`
+. `org.codice.ddf.migration.MigrationException`
+. `org.codice.ddf.migration.MigrationMetadata`
+. `org.codice.ddf.migration.MigrationWarning`
+
+===== `Migratable`
+
+The contract for a `migratable` is stored here. It is used by both sub-interfaces `ConfigurationMigratable` and `DataMigratable`.
+
+[WARNING]
+====
+Do not implement `Migratable` directly; it is intended for use only as a common base interface.
+Instead, the appropriate sub-interface should be used.
+====
+
+The `org.codice.ddf.migration.Migratable` interface defines these methods:
+
+.`MigrationMetadata export(Path exportPath) throws MigrationException`
+.`String getDescription()`
+.`boolean isOptional()`
+
+The `exportPath` in `export(Path exportPath)` is the path where all of the exportable data is copied.
+It is provided via an argument to the `migration:export` console command or via the Export Dialog in the ${admin-console}.
+The default value is `<${branding}_HOME>/etc/exported`.
+It is the responsibility of a `Migratable` to prevent naming collisions upon export.
+For example, if a `Migratable` writes files for its export, it must namespace the files.
```

Comment:
This isn't a documentation issue but I wanted to make sure it's noted...ideally, each Migratable implementation shouldn't be individually and completely responsible for namespacing its files. That's the sort of functionality I'd rather see provided by/aided by a default method.